shell/key-bindings.zsh

```diff
@@ -103,8 +103,10 @@ fzf-history-widget() {
   local ret=$?
   if [ -n "$selected" ]; then
     num=$selected[1]
-    if [ -n "$num" ]; then
+    if [[ "$num" =~ ^[0-9]+$ ]]; then
       zle vi-fetch-history -n $num
+    else # selected is a custom query, not from history
+      LBUFFER="${selected[@]}"
```

Comment:
This doesn't preserve multiple whitespace characters. e.g. `foo         bar` becomes `foo bar`

Reply:
Fixed. I turned `$selected` into plain text instead of array but not sure there is any side effect.

Reply:
Fixed. Turned $selected into plain text instead of array but i'm not sure there is any side effect.